pkg/controller/jobframework/tas.go

```
@@ -65,6 +70,16 @@ func NewPodSetTopologyRequest(meta *metav1.ObjectMeta) *podSetTopologyRequestBui
 		psTopologyReq = nil
 	}
 
+	if sliceRequiredTopologyFound && sliceSizeFound {
+		sliceSizeIntValue, err := strconv.ParseInt(sliceSizeValue, 10, 32)
+		if err != nil {
+			// silently ignore as it should not happen due to earlier validation in a webhook
```

Should we just silently ignore it, log something or return an error? 🤔 Ideally I'd log something for debugging any bugs in the future, but based on the similar code in line 67 (parsing unconstrained boolean) we silently ignore any error.

---

> Should we just silently ignore it, log something or return an error?

I think we should not silently ignore errors. Yes, currently we know this has been validated, but code evolves, and some assumptions about the two layers may change over time, leading to bugs which are very tricky to debug. For example, we relaxed validation in webhooks to support some corner case (say 0 size podsetsslices), but the code here was not ready for that.

> Ideally I'd log something for debugging any bugs in the future, but based on the similar code in line 67 (parsing unconstrained boolean) we silently ignore any error.

Agree, and I think line 67 was just an omission during review, rather than where we want to be. Thanks for raising that. 

I'm ok with a follow up to propagate both errors up, or at least log as errors.